### PR TITLE
Dev->Test (keycloak update + typing fixes)

### PR DESCRIPTION
**Pull request by brankoEISI**

### CODE REVIEW

File: packages/admin-client/src/common/components/CustomDatagrid/CustomDatagrid.tsx [hunk 1]
```diff
@@ -35,7 +35,7 @@ const CustomDatagrid = <T,>({ ariaLabel, showCalculatorButton, setIsLoading, ...
     }, [identity])
 
     useEffect(() => {
-        if (setIsLoading) {
+        if (setIsLoading && isLoading != null) {
```
Comment (makskononenko) [inline packages/admin-client/src/common/components/CustomDatagrid/CustomDatagrid.tsx:38]: Just curios can isLoading be undefined? If yes should we still perform `setIsLoading(isLoading)`?
We could make check for both null and undefined by using `!isLoading`

File: packages/admin-client/src/common/components/CustomDatagridRow/CustomDatagridRow.tsx [hunk 2]
```diff
@@ -88,7 +92,10 @@ const DatagridRow = React.forwardRef<HTMLTableRowElement, CustomDatagridRowProps
     const handleClick = useCallback(
         async (event) => {
             event.persist()
-            const type = typeof rowClick === "function" ? await rowClick(id as Identifier, resource, record) : rowClick
+            const type: string | boolean =
+                typeof rowClick === "function" && record
+                    ? await rowClick(id as Identifier, resource as string, record)
+                    : (rowClick as string)
```
Comment (makskononenko) [inline packages/admin-client/src/common/components/CustomDatagridRow/CustomDatagridRow.tsx:98]: Just curious in which case `type` can be `boolean`?  
Does `await rowClick(id as Identifier, resource as string, record)` return `string | boolean`?

It just a bit confusing to me but please feel free to ignore my comment if there is a logic behind this.

File: packages/admin-client/src/common/components/CustomDatagrid/CustomDatagrid.tsx (hunk 1 above)
Comment (brankoEISI) [reply packages/admin-client/src/common/components/CustomDatagrid/CustomDatagrid.tsx:38]: `isLoading` can be undefined. If it is we shouldnt perform `setIsLoading(isLoading)`, that's correct. The problem with `!isLoading` is that it'll return true if `isLoading` is false, as `isLoading` is a boolean value. We still want to run the logic when it's false. `isLoading != null` checks for both null and undefined, so I think it's fine here.

File: packages/admin-client/src/common/components/CustomDatagridRow/CustomDatagridRow.tsx (hunk 2 above)
Comment (brankoEISI) [reply packages/admin-client/src/common/components/CustomDatagridRow/CustomDatagridRow.tsx:98]: Yes, here's the functions type definition: `const rowClick: (id: Identifier, resource: string, record: RaRecord) => string | false | Promise<string | false>`

**Review by makskononenko — APPROVED** [review]

**Pull request merged**